python/cudf/cudf/tests/test_string.py

```diff
@@ -2932,6 +2933,144 @@ def test_string_slice_with_mask():
     assert_eq(actual, expected)
 
 
+@pytest.mark.parametrize(
+    "data",
+    [
+        [
+            """
+            {
+                "store":{
+                    "book":[
+                        {
+                            "category":"reference",
+                            "author":"Nigel Rees",
+                            "title":"Sayings of the Century",
+                            "price":8.95
+                        },
+                        {
+                            "category":"fiction",
+                            "author":"Evelyn Waugh",
+                            "title":"Sword of Honour",
+                            "price":12.99
+                        }
+                    ]
+                }
+            }
+            """
+        ],
+        [
+            """
+            {
+                "store":{
+                    "book":[
+                        {
+                            "category":"reference",
+                            "author":"Nigel Rees",
+                            "title":"Sayings of the Century",
+                            "price":8.95
+                        }
+                    ]
+                }
+            }
+            """,
+            """
+            {
+                "store":{
+                    "book":[
+                        {
+                            "category":"fiction",
+                            "author":"Evelyn Waugh",
+                            "title":"Sword of Honour",
+                            "price":12.99
+                        }
+                    ]
+                }
+            }
+            """,
+        ],
+    ],
+)
+def test_string_get_json_object(data):
+    gs = cudf.Series(data)
+    js = json.loads(data[0])
+
+    assert_eq(
+        json.loads(gs.str.get_json_object("$.store").iloc[0]), js["store"]
+    )
+    assert_eq(
+        json.loads(gs.str.get_json_object("$.store.book").iloc[0]),
+        js["store"]["book"],
+    )
+    assert_eq(
+        gs.str.get_json_object("$.store.book[0].category").iloc[0],
+        js["store"]["book"][0]["category"],
+    )
```

Comment:
Instead of directly using `json.loads(data[0])` could we apply this to a Pandas Series to replicate handling multiple elements similar to cuDF?

Reply:
Pandas doesn't have a ` get_json_object function ` (or an equivalent) for comparison purposes though?

Reply:
Yes, I'm suggesting using `pd.Series(...).apply(lambda x: json.loads(x)["store"]["book"][0]["category"])` for example.